### PR TITLE
Add cookiejar path related security issue

### PR DESCRIPTION
**Pull request by tirkarthi**
I had it in my fork and forgot to push it. bpo 30458 is also fixed but not updated. I will add them. I am opening this as WIP to make sure I don't forget it.

### CODE REVIEW

File: vulnerabilities.yaml [hunk 1]
```diff
@@ -1266,6 +1285,50 @@
...
+   - 3.8: 0e1f1f01058bd4a9b98cfe443214adecc019a38c
+  description: |
+    Cookies of example.com with path=/any where sent to example.com/anybad/
+    while using a cookiejar with http.cookiejar.DefaultCookiePolicy policy.
+    The code did not check for the first non-matching character in prefix match
+    to be a slash.
+
+- name: "CVE-2019-9947: CRLF injection attack in urllib"
+  cve: CVE-2019-9947
+  bpo: 30458
```
Comment (vstinner) [inline vulnerabilities.yaml]: This one is already tracked.

File: vulnerabilities.yaml [hunk 2]
```diff
@@ -1266,6 +1285,50 @@
     `http.cookiejar.DefaultCookiePolicy` policy.  Patch by Karthikeyan
     Singaravelan.
 
+- name: "bpo-35647: Incorrect validation of path in http.cookiejar"
+  cve:
+  bpo: 35647
+  disclosure: "2019-01-03 (Python issue bpo-35647 reported)"
+  reported-at: "2019-01-03"
+  reported-by: "Karthikeyan Singaravelan"
+  affected-versions:
```
Comment (vstinner) [inline vulnerabilities.yaml]: Remove affected-versions, it's redundant.

File: vulnerabilities.yaml [hunk 3]
```diff
@@ -21,6 +21,25 @@
 #
 # Other tag: "ignore-python3: true", used for modules removed from Python 3
 
+- name: "bpo-35755: Remove current directory from posixpath.defpath"
+  bpo: 35755
+  disclosure: "2019-01-17 (Reported by bpo-35755)"
```
Comment (vstinner) [inline vulnerabilities.yaml]: disclosure, reported-at, reported-by and affected-versions are generated automatically.

File: vulnerabilities.yaml [hunk 4]
```diff
@@ -21,6 +21,25 @@
 #
 # Other tag: "ignore-python3: true", used for modules removed from Python 3
 
+- name: "bpo-35755: Remove current directory from posixpath.defpath"
```
Comment (vstinner) [inline vulnerabilities.yaml]: Honestly, I'm not sure that it's really a security issue. If you consider that it's the case, please open a discussion on security-sig@python.org mailing list.

Comment (tirkarthi) [reply vulnerabilities.yaml]: I just added it since it was tagged as security in GitHub and bpo. I have removed it.

**Comment by tirkarthi** [comment]
I have removed the auto-generated fields mentioned. I removed the urllib and defpath items as suggested. This PR adds only the cookiejar path related vulnerability.

File: vulnerabilities.yaml [hunk 5]
```diff
@@ -1266,6 +1266,21 @@
     `http.cookiejar.DefaultCookiePolicy` policy.  Patch by Karthikeyan
     Singaravelan.
 
+- name: "bpo-35647: Incorrect validation of path in http.cookiejar"
```
Comment (vstinner) [inline vulnerabilities.yaml]: Please remove "bpo-35647: " prefix.

File: vulnerabilities.yaml [hunk 6]
```diff
@@ -1266,6 +1266,21 @@
...
+  bpo: 35647
+  fixed-in:
+   - 3.4: 382981b25092b5e9285f1e4894142af1e8f2ca86
+   - 3.5: e260f092cd0d8975c777e73ca6fb549d59b5d452
+   - 3.6: 5565b1db6f37f244890369e0d68a3e906aca28b9
+   - 3.7: 97c7d78fda49e03fc773c171ce0c736d02bb73f5
+   - 3.8: 0e1f1f01058bd4a9b98cfe443214adecc019a38c
+  description: |
+    Cookies of example.com with path=/any where sent to example.com/anybad/
+    while using a cookiejar with http.cookiejar.DefaultCookiePolicy policy.
```
Comment (vstinner) [inline vulnerabilities.yaml]: Please use reST syntax here for better rendering:

```
    Cookies of ``example.com`` with ``path=/any`` where sent to ``example.com/anybad/``
    while using a cookiejar with ``http.cookiejar.DefaultCookiePolicy`` policy.
```

**Comment by vstinner** [comment]
Thanks, merged. It's good to see more contributors on this project ;-) I was feeling alone :-D

@tirkarthi: Can you try to add a script to detect if Python is vulnerable to bpo-35647 in check-python-vuln//check-python-vuln.py?

**Comment by tirkarthi** [comment]
Thanks Victor for merging it. Sure, I will add a script as another PR.

**Pull request merged**